plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -838,6 +863,119 @@ public Optional<JdbcTableHandle> getTableHandle(ConnectorSession session, Schema
                         "com.google.common.util.concurrent.UncheckedExecutionException: java.lang.RuntimeException: first attempt is poised to fail");
     }
 
+    @Test
+    public void testSpecificSchemaAndTableCachesWithCacheMissing()
+            throws Exception
+    {
+        assertSpecificSchemaAndTableCaches(true);
+    }
+
+    @Test
+    public void testSpecificSchemaAndTableCachesWithoutCacheMissing()
+            throws Exception
+    {
+        assertSpecificSchemaAndTableCaches(false);
+    }
+
+    private void assertSpecificSchemaAndTableCaches(boolean cacheMissing)
```

Comment:
Can we split this into two methods - one for table and other for schema ?

Reply:
The test ensures that schema and table names cache ttls are different, it won't do that if they are separate.

Reply:
The test method is quite big and it does assert multiple operations like with and without cache. Or we could have schema name cache, table name cache and a test method with combined configuration.

Reply:
Those calls are inter-dependendent. First of all the test ensure that schema and table caches are different, an then it tests the evolution of the cache over time. I can split it into separate 'stages' which should make it more readable.

Reply:
Split it into multiple smaller sections and extracted to a separate assertions class

Reply:
Not sure if the two values for boolean warrant a DataProvider, and it's not really data, it's two modes of operations and they should be expressed explcitly.